src/v/cluster/self_test/tests/self_test_request_test.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "bytes/iobuf.h"
+#include "cluster/self_test_rpc_types.h"
+
+#include <seastar/testing/thread_test_case.hh>
+
+SEASTAR_THREAD_TEST_CASE(test_make_netcheck_request) {
+    static const auto check_frags =
+      [](iobuf::const_iterator begin, const iobuf::const_iterator& end) {
+          while (begin != end) {
+              if (begin->size() != 8192) {
+                  return false;
+              }
+              begin++;
+          }
+          return true;
+      };
+
+    auto req = cluster::make_netcheck_request(model::node_id{0}, 52).get0().buf;
+    BOOST_CHECK_EQUAL(req.size_bytes(), 52);
+    BOOST_CHECK_EQUAL(std::distance(req.cbegin(), req.cend()), 1);
+    BOOST_CHECK_EQUAL(req.cbegin()->size(), 52);
+    req = cluster::make_netcheck_request(model::node_id{0}, 8192).get0().buf;
+    BOOST_CHECK_EQUAL(req.size_bytes(), 8192);
+    BOOST_CHECK_EQUAL(std::distance(req.cbegin(), req.cend()), 1);
+    BOOST_CHECK_EQUAL(req.cbegin()->size(), 8192);
+    req = cluster::make_netcheck_request(model::node_id{0}, 8193).get0().buf;
+    BOOST_CHECK_EQUAL(req.size_bytes(), 8193);
+    BOOST_CHECK_EQUAL(std::distance(req.cbegin(), req.cend()), 2);
+    BOOST_CHECK_EQUAL(req.cbegin()->size(), 8192);
+    BOOST_CHECK_EQUAL((++req.cbegin())->size(), 1);
+    const auto size = ((8192 * 52) + 100);
+    req = cluster::make_netcheck_request(model::node_id{0}, size).get0().buf;
+    BOOST_CHECK_EQUAL(req.size_bytes(), size);
+    BOOST_CHECK_EQUAL(std::distance(req.cbegin(), req.cend()), 53);
+    BOOST_CHECK(check_frags(req.cbegin(), (--(--req.cend()))));
```

Comment:
Why the double --? Doesn't `check_frags()` skip comparing `end`?

Reply:
No, but I think modifying `check_frags()` to make it more obvious to whats going on is the move